go-controller/pkg/util/ovs.go

```diff
@@ -70,76 +63,87 @@ func runningPlatform() (string, error) {
 	return "", fmt.Errorf("Unknown platform")
 }
 
-// RunOVSOfctl runs a command via ovs-ofctl.
-func RunOVSOfctl(args ...string) (string, string, error) {
-	cmdPath, err := exec.LookPath(ovsOfctlCommand)
+// Exec runs various OVN and OVS utilities
+type execHelper struct {
+	exec      kexec.Interface
+	ofctlPath string
+	vsctlPath string
+	nbctlPath string
+	ipPath    string
+}
+
+var runner *execHelper
+
+// SetExec validates executable paths and saves the given exec interface
+// to be used for running various OVS and OVN utilites
+func SetExec(exec kexec.Interface) error {
+	var err error
+
+	runner = &execHelper{exec: exec}
+	runner.ofctlPath, err = exec.LookPath(ovsOfctlCommand)
+	if err != nil {
+		return err
+	}
+	runner.vsctlPath, err = exec.LookPath(ovsVsctlCommand)
 	if err != nil {
-		return "", "", err
+		return err
 	}
+	runner.nbctlPath, err = exec.LookPath(ovnNbctlCommand)
+	if err != nil {
+		return err
+	}
+	runner.ipPath, err = exec.LookPath(ipCommand)
```

Comment:
This does not work on Windows. ovnkube fails windows clustering because of this.